commit 058859f9dd9860c5422cdcf48ca70e2ecde06854
Author: Ilan Karasik <ilankarasik@gmail.com>
Date:   Thu Mar 3 18:15:44 2016 -0500

    Changed browser.syncui.js

diff --git a/browser/base/content/browser-syncui.js b/browser/base/content/browser-syncui.js
index 4a3c63b..4dc5cb6 100644
--- a/browser/base/content/browser-syncui.js
+++ b/browser/base/content/browser-syncui.js
@@ -39,10 +39,7 @@ var gSyncUI = {
 
     // Proceed to set up the UI if Sync has already started up.
     // Otherwise we'll do it when Sync is firing up.
-    if (this.weaveService.ready) {
-      this.initUI();
-      return;
-    }
+    this.weaveService.whenLoaded().then(() => this.initUI())
 
     // Sync isn't ready yet, but we can still update the UI with an initial
     // state - we haven't called initUI() yet, but that's OK - that's more
